src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -457,27 +457,55 @@ func (r *StatefulSetResource) obj() (k8sclient.Object, error) {
 	return ss, nil
 }
 
-func overprovisioned(developerMode bool, limits corev1.ResourceList) []string {
-	memory, exist := limits["memory"]
-	if !exist {
-		memory = resource.MustParse("2Gi")
-	}
+func prepareAdditionalArguments(
+	developerMode bool, originalRequests corev1.ResourceList,
+) []string {
+	requests := originalRequests.DeepCopy()
+
+	requests.Cpu().RoundUp(0)
+	requestedCores := requests.Cpu().Value()
+	requestedMemory := requests.Memory().Value()
 
 	if developerMode {
-		return []string{
+		args := []string{
 			"--overprovisioned",
 			// sometimes a little bit of memory is consumed by other processes than seastar
 			"--reserve-memory " + redpandav1alpha1.ReserveMemoryString,
-			"--smp=1",
 			"--kernel-page-cache=true",
 			"--default-log-level=debug",
 		}
+		// When smp is not set, all cores are used
+		if requestedCores > 0 {
+			args = append(args, "--smp="+strconv.FormatInt(requestedCores, 10))
+		}
+		return args
 	}
-	return []string{
+
+	args := []string{
 		"--default-log-level=info",
 		"--reserve-memory 0M",
-		"--memory " + strconv.FormatInt(memory.Value(), 10),
 	}
+
+	/*
+	 * Example:
+	 *    in: minimum requirement per core, 2GB
+	 *    in: requestedMemory, 10GB
```

Comment:
requestedMemory = 16G?

how can 10GB/8 cores == 2G/core?

Reply:
Thanks, yeah that was meant to be 16GB. I fixed it.